pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+
+        dataGroup = dataGroups[1]
+        for prop in dataGroup.findall('property'):
+            if prop.attrib['name'] == 'time':
+                time_dict = {'units': prop.attrib['units'], 'column': 0}
+                time_tag = prop.attrib['id']
+            elif prop.attrib['name'] == 'volume':
+                volume_dict = {'units': prop.attrib['units'], 'column': 1}
+                volume_tag = prop.attrib['id']
+
+        volume_history = {'time': time_dict, 'volume': volume_dict, 'values': []}
+
+        # collect volume-time history
+        for dp in dataGroup.findall('dataPoint'):
+            time = None
+            volume = None
+            for val in dp:
+                if val.tag == time_tag:
+                    time = float(val.text)
+                elif val.tag == volume_tag:
+                    volume = float(val.text)
+            volume_history['values'].append([time, volume])
+
+        datapoints[0]['volume-history'] = volume_history
+
+    elif len(dataGroups) > 2:
+        raise NotImplementedError('More than two DataGroups not supported.')
+
+    return datapoints
+
+
+def convert_ReSpecTh(filename_xml, output='', file_author='', file_author_orcid=''):
+    """Convert ReSpecTh XML file to ChemKED YAML file.
+
+    Args:
+        filename_xml (`str`): Name of ReSpecTh XML file to be converted.
+        output (`str`, optional): Output path for converted file.
+        file_author (`str`, optional): Name to override original file author
+        file_author_orcid (`str`, optional): ORCID of file author
+
+    Returns:
+        filename_yaml (`str`): Name of newly created ChemKED YAML file.
+    """
+    assert os.path.isfile(filename_xml), 'Error: ' + filename_xml + ' file missing'
+
+    # get all information from XML file
+    try:
+        tree = etree.parse(filename_xml)
+    except OSError:
+        raise OSError('Unable to open file ' + filename_xml)
+    root = tree.getroot()
+
+    # get file metadata
+    properties = get_file_metadata(root)
+
+    # get reference info
+    properties['reference'] = get_reference(root)
+    # Save name of original data filename
+    if properties['reference'].get('detail') is None:
+        properties['reference']['detail'] = ''
+    properties['reference']['detail'] += ('Converted from XML file ' +
+                                          os.path.basename(filename_xml)
+                                          )
+
+    # Ensure ignition delay, and get which kind of experiment
+    properties.update(get_experiment_kind(root))
+
+    # Get properties shared across the file
+    properties['common-properties'] = get_common_properties(root)
+
+    # Determine definition of ignition delay
+    properties['common-properties']['ignition-type'] = get_ignition_type(root)
+
+    # Now parse ignition delay datapoints
+    properties['datapoints'] = get_datapoints(root)
+
+    # Get compression time for RCM, if volume history given
```

Comment:
Yeah, this isn't necessary anyways, I think—I believe I intended on coming back to this, but I'll just remove it altogether.